server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -111,6 +118,142 @@ private Optional<String> validateDateOfBirth(String dob) {
     return Optional.empty();
   }
 
+  private Form<EditTiClientInfoForm> validateFirstNameForEditClient(
+      Form<EditTiClientInfoForm> form) {
+    if (Strings.isNullOrEmpty(form.value().get().getFirstName())) {
+      return form.withError(FORM_FIELD_NAME_FIRST_NAME, "First name required");
+    }
+    return form;
+  }
+
+  private Form<EditTiClientInfoForm> validateLastNameForEditClient(
+      Form<EditTiClientInfoForm> form) {
+    if (Strings.isNullOrEmpty(form.value().get().getLastName())) {
+      return form.withError(FORM_FIELD_NAME_LAST_NAME, "Last name required");
+    }
+    return form;
+  }
+
+  private Boolean hasEmailChanged(String newEmail, AccountModel account) {
+    return !newEmail.equals(account.getEmailAddress());
+  }
+
+  private Form<EditTiClientInfoForm> validatePhoneNumber(Form<EditTiClientInfoForm> form) {
+    String phoneNumber = form.value().get().getPhoneNumber();
+    if (!Strings.isNullOrEmpty(phoneNumber)) {
+      phoneNumber = phoneNumber.replaceAll("[^0-9]", "");
+      if (phoneNumber.length() != 10) {
+        return form.withError(FORM_FIELD_NAME_PHONE, "A phone number must contain 10 digits");
+      }
+      if (!phoneNumber.matches("[0-9]+")) {
+        return form.withError(FORM_FIELD_NAME_PHONE, "A phone number must contain only digits");
+      }
+      try {
+        Phonenumber.PhoneNumber phonenumber = PHONE_NUMBER_UTIL.parse(phoneNumber, "US");
+        if (!PHONE_NUMBER_UTIL.isValidNumber(phonenumber)) {
+          return form.withError(FORM_FIELD_NAME_PHONE, "This phone number is not valid");
+        }
+      } catch (NumberParseException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    return form;
+  }
+
+  private Form<EditTiClientInfoForm> validateDateOfBirth(Form<EditTiClientInfoForm> form) {
+    Optional<String> errorMessage = validateDateOfBirth(form.value().get().getDob());
+    if (errorMessage.isPresent()) {
+      return form.withError(FORM_FIELD_NAME_DOB, errorMessage.get());
+    }
+    return form;
+  }
+
+  private Form<EditTiClientInfoForm> validateEmailAddress(
+      Form<EditTiClientInfoForm> form, AccountModel currentAccount) {
+    String newEmail = form.get().getEmailAddress();
+    // email addresses not a requirement for TI Client
+    if (Strings.isNullOrEmpty(newEmail)) {
+      return form;
+    }
+    if (hasEmailChanged(newEmail, currentAccount)
+        && accountRepository.lookupAccountByEmail(newEmail).isPresent()) {
```

Comment:
_Would there be an error if the TI tried to create a client with the email janedoe@email.com?_ - Yes they cannot create a client who already has an account associated with existing email. But the good thing is email address is not a requirement for TI Client, So Jane can still get the benefits, provided, she doesn't use any email address.

Reply:
I think this check is always needed. One one case it could be Jane Doe wanting help after she started an application but in another case it could be a malicious user who impersonates Jane Doe. All they need is Jane Doe's email address, and suddenly, they can see all of Jane Doe's Application and can take a print of it.

Reply:
Oh I see, so without this check then a malicious TI could put in someone's email and see all their applications! I didn't think of that, thanks for that clarification.